lib/Dialect/FIRRTL/Transforms/ExtractInstances.cpp

```diff
@@ -695,12 +711,15 @@ void ExtractInstancesPass::extractInstances() {
         // its path. If that is the case we have to convert the NLA into a
         // regular local annotation.
         if (nlaPath.size() == 2) {
-          anno.removeMember("circt.nonlocal");
-          newInstAnnos.push_back(anno.getDict());
+          auto nlaAnno = instNonlocalAnnos.find(nla);
+          if (nlaAnno != instNonlocalAnnos.end()) {
+            auto anno = nlaAnno->getSecond();
+            anno.removeMember("circt.nonlocal");
+            newInstNonlocalAnnos.push_back(anno);
+          }
+          nlaTable.erase(nla);
           nlasToRemove.insert(nla);
-          LLVM_DEBUG(llvm::dbgs()
-                     << "    - Converted to local " << anno.getDict() << "\n");
```

Comment:
Would it make sense to move this debug print into the if condition with `nlaAnno` such that you can see NLA to local anno conversion in debug runs (like the "- Created ..." output above)?